src/Disks/StoragePolicy.cpp

```diff
@@ -56,13 +57,31 @@ StoragePolicy::StoragePolicy(
         config.keys(volumes_prefix, keys);
     }
 
+    std::unordered_set<UInt64> volume_priorities;
+
     for (const auto & attr_name : keys)
     {
         if (!std::all_of(attr_name.begin(), attr_name.end(), isWordCharASCII))
             throw Exception(ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG,
                             "Volume name can contain only alphanumeric and '_' in storage policy {} ({})",
                             backQuote(name), attr_name);
         volumes.emplace_back(createVolumeFromConfig(attr_name, config, volumes_prefix + "." + attr_name, disks));
+        if (volumes.back()->volume_priority != std::numeric_limits<UInt64>::max())
+            volume_priorities.insert(volumes.back()->volume_priority);
+    }
+
+    if (!volume_priorities.empty())
+    {
+        /// Check that priority values cover the range from 1 to N (lowest explicit priority)
+        if (*std::min_element(volume_priorities.begin(), volume_priorities.end()) != 1
+            || *std::max_element(volume_priorities.begin(), volume_priorities.end()) != volume_priorities.size())
+            throw Exception(
+                ErrorCodes::INVALID_CONFIG_PARAMETER,
+                "volume_priority values must cover the range from 1 to N (lowest priority specified) without gaps");
```

Comment:
At first glance priorities 1,1,4,4 would be considered valid.

Reply:
At the second glance priorities 1,1,2,2 would be considered valid ;)
Do we check if we have non-unique values?